aten/src/ATen/native/MaxPooling.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/native/DispatchStub.h>
+
+namespace at {
+namespace native {
+
+// TODO(Heitor) Template by dimension
+struct PoolingParams1D {
+  int64_t NB; // Number of batches
+  int64_t NC; // Number of channels
+  int64_t IW; // Input width
+  int64_t OW; // Output width
+  int64_t KW; // Kernel width
+  int64_t SJ; // Column stride
+  int64_t PJ; // Column padding
+  int64_t DJ; // Column dilation
+
+  // Return index of first output within bounds for this kernel index
+  inline int64_t valid_kernel_start(int64_t kj) const {
+    int64_t ij = kj * DJ - PJ;
+    return ij < 0 ? (-ij + SJ - 1) / SJ : 0;
```

Comment:
nit at::divup?

Reply:
I was thinking about using that but it's defined in Parallel.h and I don't really have anything parallel related here. So I was thinking if we should move at::divup somewhere more general?

Reply:
I think divup is added for parallel code first, that's why it is there. Since the namespace is at, it doesn't matter where it is I think. We do have a math.h somewhere though.